Extend G0 ADC features

victorandrehc opened the pull request:
1. Adds IRQ and DMA capabilities for the G0 implementation. 
2. Adds Vbat, Vref and temperature channel bits on the ADC.

victorandrehc committed:
[stm32] Extend Timer features: PWM modes and Comparator

Javask committed:
[cmake] Change template to respect repo name

Javask committed:
[cmake] Support header only libraries

Javask committed:
[cmake] Correctly propagate options to all repos

Javask committed:
[cmake] Emulate scons behaviour of including everything

Javask committed:
[cmake] Fix build on windows

wald-zat committed:
[stm32] i2c irq priority can be set

klsc-zeat commented on src/modm/platform/adc/stm32f0/adc_impl.hpp.in:
```diff
@@ -360,3 +366,103 @@ modm::platform::Adc{{ id }}::setWaitMode(const bool enable)
+}
+
+bool
+	modm::platform::Adc{{ id }}::getAdcEnabled()
```
whole file has funny indentation, but this seems wrong

klsc-zeat commented:
Squash all commits before generating PR to modm?

klsc-zeat commented on examples/nucleo_g071rb/adc_dma/main.cpp:
```diff
@@ -0,0 +1,97 @@
+
+int main()
+{
+   Board::initialize();
```
example tested and works?

victorandrehc replied:
The example is fixed.

Commit:
[stm32] add advanced initialize on iwdg

Commit:
[stm32] add enable break in timer

victorandrehc commented:
PR created on modm mainstream

hshose committed:
[driver] Adding AS5047 driver and example

TomSaw committed:
fix clock in lis3::SpiMaster::initialize

Javask committed:
[cmake] Update auto-generated CMakeLists

chris-durand committed:
[stm32] Fix flash page size for large F103, F105 and F107 devices

hshose committed:
[ext] example for json usage added

rleh committed:
[stm32/sam_x7x] Allow I2C ISR priority to be set

rleh committed:
Fix Python syntax

Since Python 3.6 backslash-character pairs that are not valid escape sequences are deprecated and generate a 'SyntaxWarning: invalid escape sequence' warning.
This is fixed by using raw string.

rleh committed:
[ext] Update modm-devices submodule

hshose committed:
[cortex-m] added flash_reserved section

Co-authored-by: Niklas Hauser <niklas.hauser@rwth-aachen.de>

hshose committed:
[ext] added nlohmann json submodule

kapacuk committed:
[tools] Improvements to BlackMagicProbe tool

- Added a missing import to `modm_tools/bmp.py`
- Changed `guess_serial_port()` to use `/dev/ttyBmpGdb` when guessing a BMP port

hshose committed:
[flash] STM32G4 flash erase dual bank BKER bit

Co-authored-by: Niklas Hauser <niklas.hauser@rwth-aachen.de>

salkinium committed:
[ext] Update CMSIS headers for STM32

salkinium committed:
[lbuild] distutils is deprecated since Python 3.12

becseya committed:
[tools] Font generation with Python 3

salkinium committed:
[release] Update changelog for 2024q1 release

Remove deprecated statements

el-han committed:
[cortex-m] Communicate the flash_offset to the dfu-util tool

wald-zat committed:
[stm32] Add DMA capabilities to F0/G0 ADC

Co-authored-by: Christopher Durand <christopher.durand@rwth-aachen.de>

wald-zat committed:
[examples] Add ADC with DMA example for STM32F0

Co-authored-by: Christopher Durand <christopher.durand@rwth-aachen.de>
